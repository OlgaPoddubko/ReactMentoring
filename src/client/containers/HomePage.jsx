// @flow
import * as React from 'react';
import { connect } from 'react-redux';

import ErrorBoundary from '../components/ErrorBoundary';
import SearchBlock from '../components/SearchBlock';
import ResultsSort from '../components/ResultsSort';
import MovieList from '../components/MovieList';
import EmptySearchResult from '../components/EmptySearchResult';
import Footer from '../components/Footer';

type Props = {
  loading: boolean,
  movies: [],
}

class HomePage extends React.Component<Props> {
  static defaultProps = {
    loading: false,
    movies: [],
  };

  render() {
    const { loading, movies } = this.props;
    return (
      <React.Fragment>
        <ErrorBoundary >
          <SearchBlock />
          {loading ?
            <div>Loading...</div>
            : movies.length ?
          <React.Fragment>
            <ResultsSort />
            <MovieList/>
          </React.Fragment>
          : <EmptySearchResult />
          }
          <Footer />
        </ErrorBoundary>
      </React.Fragment>
    );
  }
}

const mapStateToProps = state => ({
  loading: state.gallery.loading,
  movies: state.gallery.movies,
});

export default connect(mapStateToProps)(HomePage);
