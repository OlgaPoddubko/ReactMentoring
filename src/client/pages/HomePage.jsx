import React from 'react';
import movieArr from '../movieArr';

import MovieList from '../components/MovieList';

class HomePage extends React.Component {
  constructor(props) {
    super(props);
    this.state = { data: movieArr };
  }
  render() {
    return <MovieList movies={this.state.data} />;
  }
}

export default HomePage;
