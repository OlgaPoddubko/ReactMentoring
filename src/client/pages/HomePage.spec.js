import React from 'react';
import HomePage from './HomePage';

describe('render HomePage', () => {
  it('should render HomePage', () => {
    const wrapper = shallow(<HomePage />);
    expect(wrapper).toMatchSnapshot();
  });
});
