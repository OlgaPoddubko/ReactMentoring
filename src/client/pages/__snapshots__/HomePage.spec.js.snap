// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render HomePage should render HomePage 1`] = `
<React.Fragment>
  <ErrorBoundary>
    <SearchBlock />
    <ResultsSort
      total={3000}
    />
    <MovieList
      movies={
        Array [
          Object {
            "budget": 28000000,
            "genres": Array [
              "Romance",
              "Horror",
              "Comedy",
              "Thriller",
            ],
            "id": 58431,
            "overview": "A zombie outbreak has fallen upon the land in this reimagining of Jane Austen’s classic tale of the tangled relationships between lovers from different social classes in 19th century England. Feisty heroine Elizabeth Bennet (Lily James) is a master of martial arts and weaponry and the handsome Mr. Darcy (Sam Riley) is a fierce zombie killer, yet the epitome of upper class prejudice. As the zombie outbreak intensifies, they must swallow their pride and join forces on the blood-soaked battlefield in order to conquer the undead once and for all.",
            "poster_path": "https://image.tmdb.org/t/p/w500/jsJlOhVHiz1xXr36616EI8tJ4b2.jpg",
            "release_date": "2016-02-04",
            "revenue": 16374328,
            "runtime": 108,
            "tagline": "Bloody lovely.",
            "title": "Pride and Prejudice and Zombies",
            "vote_average": 5.5,
            "vote_count": 826,
          },
          Object {
            "budget": 28000000,
            "genres": Array [
              "Drama",
              "Romance",
            ],
            "id": 4348,
            "overview": "Pride & Prejudice is a humorous story of love and life among English gentility during the Georgian era. Mr. Bennet is an English gentleman living in Hertfordshire with his overbearing wife and five daughters. If Mr. Bennet dies their house will be inherited by a distant cousin whom they have never met, so the family's future happiness and security is dependent on the daughters making good marriages.",
            "poster_path": "https://image.tmdb.org/t/p/w500/lAb9l4kgc6QWnHamBzTnskt71A7.jpg",
            "release_date": "2005-09-16",
            "revenue": 121147947,
            "runtime": 135,
            "tagline": "A romance ahead of its time.",
            "title": "Pride & Prejudice",
            "vote_average": 7.8,
            "vote_count": 1901,
          },
          Object {
            "budget": 0,
            "genres": Array [
              "Adventure",
              "Animation",
              "Family",
            ],
            "id": 9732,
            "overview": "The circle of life continues for Simba, now fully grown and in his rightful place as the king of Pride Rock. Simba and Nala have given birth to a daughter, Kiara who's as rebellious as her father was. But Kiara drives her parents to distraction when she catches the eye of Kovu, the son of the evil lioness, Zira. Will Kovu steal Kiara's heart?",
            "poster_path": "https://image.tmdb.org/t/p/w500/9FHNYvpltLQfpakXpMMj7RsVHqm.jpg",
            "release_date": "1998-10-24",
            "revenue": 0,
            "runtime": 81,
            "tagline": "The Circle of Life Continues...",
            "title": "The Lion King 2: Simba's Pride",
            "vote_average": 6.7,
            "vote_count": 1412,
          },
          Object {
            "budget": 19500000,
            "genres": Array [
              "Drama",
              "Fantasy",
              "Romance",
            ],
            "id": 399055,
            "overview": "An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.",
            "poster_path": "https://image.tmdb.org/t/p/w500/k4FwHlMhuRR5BISY2Gm2QZHlH5Q.jpg",
            "release_date": "2017-12-01",
            "revenue": 185545281,
            "runtime": 135,
            "tagline": "A Fairy Tale for Troubled Times",
            "title": "The Shape of Water",
            "vote_average": 7.3,
            "vote_count": 3200,
          },
          Object {
            "budget": 150000000,
            "genres": Array [
              "Animation",
              "Adventure",
              "Family",
              "Comedy",
            ],
            "id": 269149,
            "overview": "Determined to prove herself, Officer Judy Hopps, the first bunny on Zootopia's police force, jumps at the chance to crack her first case - even if it means partnering with scam-artist fox Nick Wilde to solve the mystery.",
            "poster_path": "https://image.tmdb.org/t/p/w500/sM33SANp9z6rXW8Itn7NnG1GOEs.jpg",
            "release_date": "2016-02-11",
            "revenue": 1023784195,
            "runtime": 135,
            "tagline": "Welcome to the urban jungle.",
            "title": "Zootopia",
            "vote_average": 7.7,
            "vote_count": 6795,
          },
          Object {
            "budget": 200000000,
            "genres": Array [
              "Fantasy",
              "Adventure",
              "Science Fiction",
            ],
            "id": 181808,
            "overview": "Rey develops her newly discovered abilities with the guidance of Luke Skywalker, who is unsettled by the strength of her powers. Meanwhile, the Resistance prepares to do battle with the First Order.",
            "poster_path": "https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg",
            "release_date": "2017-12-13",
            "revenue": 1325937250,
            "runtime": 135,
            "tagline": "The Saga Continues",
            "title": "Star Wars: The Last Jedi",
            "vote_average": 7.1,
            "vote_count": 4732,
          },
          Object {
            "budget": 200000000,
            "genres": Array [
              "Action",
              "Adventure",
              "Fantasy",
              "Science Fiction",
            ],
            "id": 284054,
            "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan \\"special forces\\"), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
            "poster_path": "https://image.tmdb.org/t/p/w500/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
            "release_date": "2018-02-13",
            "revenue": 1245257672,
            "runtime": 135,
            "tagline": "Long live the king",
            "title": "Black Panther",
            "vote_average": 7.3,
            "vote_count": 3788,
          },
          Object {
            "budget": 175000000,
            "genres": Array [
              "Adventure",
              "Comedy",
              "Family",
              "Animation",
            ],
            "id": 354912,
            "overview": "Despite his family’s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel's family history.",
            "poster_path": "https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
            "release_date": "2017-10-27",
            "revenue": 700920729,
            "runtime": 135,
            "tagline": "The celebration of a lifetime",
            "title": "Coco",
            "vote_average": 7.8,
            "vote_count": 3619,
          },
          Object {
            "budget": 175000000,
            "genres": Array [
              "Adventure",
              "Science Fiction",
              "Action",
            ],
            "id": 333339,
            "overview": "When the creator of a popular video game system dies, a virtual contest is created to compete for his fortune.",
            "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
            "release_date": "2018-03-28",
            "revenue": 0,
            "runtime": 135,
            "tagline": "A better reality awaits.",
            "title": "Ready Player One",
            "vote_average": 8.1,
            "vote_count": 617,
          },
          Object {
            "budget": 94000000,
            "genres": Array [
              "Action",
              "Adventure",
            ],
            "id": 338970,
            "overview": "Lara Croft, the fiercely independent daughter of a missing adventurer, must push herself beyond her limits when she finds herself on the island where her father disappeared.",
            "poster_path": "https://image.tmdb.org/t/p/w500/ePyN2nX9t8SOl70eRW47Q29zUFO.jpg",
            "release_date": "2018-03-08",
            "revenue": 126025000,
            "runtime": 135,
            "tagline": "Her legend begins",
            "title": "Tomb Raider",
            "vote_average": 6.2,
            "vote_count": 817,
          },
          Object {
            "budget": 180000000,
            "genres": Array [
              "Action",
              "Adventure",
              "Fantasy",
            ],
            "id": 284053,
            "overview": "Thor is imprisoned on the other side of the universe and finds himself in a race against time to get back to Asgard to stop Ragnarok, the prophecy of destruction to his homeworld and the end of Asgardian civilization, at the hands of an all-powerful new threat, the ruthless Hela.",
            "poster_path": "https://image.tmdb.org/t/p/w500/rzRwTcFvttcN1ZpX2xv4j3tSdJu.jpg",
            "release_date": "2017-10-25",
            "revenue": 854229371,
            "runtime": 135,
            "tagline": "No Hammer. No Problem.",
            "title": "Thor: Ragnarok",
            "vote_average": 7.4,
            "vote_count": 5349,
          },
          Object {
            "budget": 160000000,
            "genres": Array [
              "Family",
              "Fantasy",
              "Romance",
            ],
            "id": 321612,
            "overview": "A live-action adaptation of Disney's version of the classic tale of a cursed prince and a beautiful young woman who helps him break the spell.",
            "poster_path": "https://image.tmdb.org/t/p/w500/tWqifoYuwLETmmasnGHO7xBjEtt.jpg",
            "release_date": "2017-03-16",
            "revenue": 1263521126,
            "runtime": 135,
            "tagline": "Be our guest.",
            "title": "Beauty and the Beast",
            "vote_average": 6.8,
            "vote_count": 7861,
          },
        ]
      }
    />
    <Footer />
  </ErrorBoundary>
</React.Fragment>
`;
